internal/host/validator.go

```diff
@@ -864,3 +864,76 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
+	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return ValidationSuccess
+	}
+	apiDomainName := fmt.Sprintf("api.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+
+	return checkDomainNameResolution(c, apiDomainName)
+}
+
+func (v *validator) printIsAPIDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
+	apiDomainName := fmt.Sprintf("api.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+	return printIsDomainNameResolvedCorrectly(c, status, apiDomainName)
+}
+
+func (v *validator) isAPIInternalDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return ValidationSuccess
+	}
+	apiInternalDomainName := fmt.Sprintf("api-int.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+	return checkDomainNameResolution(c, apiInternalDomainName)
+}
+
+func (v *validator) printIsAPIInternalDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
+	apiInternalDomainName := fmt.Sprintf("api-int.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+	return printIsDomainNameResolvedCorrectly(c, status, apiInternalDomainName)
+}
+
+func (v *validator) isAppsDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return ValidationSuccess
+	}
+	appsDomainName := fmt.Sprintf("console-openshift-console.apps.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+	return checkDomainNameResolution(c, appsDomainName)
+}
+
+func (v *validator) printIsAppsDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
+	appsDomainName := fmt.Sprintf("*.apps.%s.%s", c.cluster.Name, c.cluster.BaseDNSDomain)
+	return printIsDomainNameResolvedCorrectly(c, status, appsDomainName)
+}
+
+func checkDomainNameResolution(c *validationContext, domainName string) ValidationStatus {
+	var response *models.DomainResolutionResponse
+
+	if err := json.Unmarshal([]byte(c.host.DomainNameResolutions), &response); err != nil {
+		return ValidationError
+	}
+
+	for _, domain := range response.Resolutions {
+		if *domain.DomainName == domainName {
+			if len(domain.IPV4Addresses) != 0 || len(domain.IPV6Addresses) != 0 {
+				return ValidationSuccess
+			}
+		}
+	}
+	return ValidationFailure
+}
+
+func printIsDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus, domainName string) string {
+	switch status {
+	case ValidationSuccess:
+		if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+			return "Domain name resolution is not requierd: non User Managed Network"
+		}
+		return fmt.Sprintf("Domain name resolution was successful for domain %s", domainName)
+	case ValidationFailure:
+		return fmt.Sprintf("Failed to resolve domain name %s correctly on the host. Expected name to resolve to something. The installation will not be able to complete as long as the DNS resolution is empty. Please create the necessary DNS entries to continue.", domainName)
```

Comment:
I think we might want more specific messages here for each validation. For the API domain, we should tell the user they should configure a DNS entry to point that domain to <their OpenShift API load balancer / one of the host interfaces temporarily>.

For the apps case, they might want to point it at ingress, etc.

Maybe give them a link to documentation about this topic so they understand better.

@avishayt / @ronniel1 / @nirfarkas

Reply:
As discussed in [here](https://coreos.slack.com/archives/CUPJTHQ5P/p1626778595210400) we will change the messages in a future commit.

Reply:
https://issues.redhat.com/browse/MGMT-7324